interface/reports/inventory_list.php

```diff
@@ -26,46 +26,662 @@
+        "FROM drug_sales AS ds, drug_inventory AS di " .
+        "LEFT JOIN list_options AS lo ON lo.list_id = 'warehouse' AND " .
+        "lo.option_id = di.warehouse_id AND lo.activity = 1 " .
+        "WHERE " .
+        "ds.sale_date >= ? AND " .
+        "di.inventory_id = ds.xfer_inventory_id " .
+        "$prodcond $extracond" .
+        "GROUP BY ds.sale_date ORDER BY ds.sale_date",
+        array($begdate)
+    );
+    while ($row = sqlFetchArray($res)) {
+        $thisdate = $row['sale_date'];
+        if (!isset($qtys[$thisdate])) {
+            $qtys[$thisdate] = 0;
+        }
+        $qtys[$thisdate] -= $row['quantity'];
+    }
+
+    // Sort by reverse date.
+    krsort($qtys);
+
+    $lastdate = '';
+    $lastqty = $current_qoh;
+
+    // This will be the count of days that have zero quantity at the start of the day.
+    $zerodays = 0;
+
+    // Now we traverse the array in descending date order, adding a date's quantity adjustment
+    // to the running total to get the quantity at the beginning of that date.
+    foreach ($qtys as $key => $val) {
+        if ($lastdate && $lastqty < $min_sale) {
+            // The span of days from $key to start of $lastdate has zero quantity.
+            // Add that number of days to $zerodays.
+            $diff = date_diff(date_create($key), date_create($lastdate));
+            $zerodays += $diff->days;
+        }
+        $lastdate = $key;
+        $lastqty += $val; // giving qoh at the start of $lastdate
+    }
+    // The last array entry hasn't been accounted for yet, so do that.
+    if ($lastqty < $min_sale) {
+        ++$zerodays;
+    }
+    return $zerodays;
+}
+
+function write_report_line(&$row)
+{
+    global $form_details, $wrl_last_drug_id, $warnings, $encount, $fwcond, $uwcond, $form_days;
+    global $gbl_expired_lot_warning_days, $form_facility, $form_warehouse, $form_action;
+
+    $emptyvalue = $form_action != 'export' ? '&nbsp;' : '';
+    $drug_id = 0 + $row['drug_id'];
+    $on_hand = 0 + $row['on_hand'];
+    $warehouse_id = isset($row['warehouse_id']) ? $row['warehouse_id'] : '';
+    $facility_id = empty($row['option_value']) ? '0' : $row['option_value'];
+    $warnings = '';
+
+    // Get sales in the date range for this drug (and facility or warehouse if details).
+    if ($form_details == 1) { // facility details
+        $query = "SELECT " .
+            "SUM(s.quantity) AS sale_quantity " .
+            "FROM drug_sales AS s " .
+            "LEFT JOIN drug_inventory AS di ON di.inventory_id = s.inventory_id " .
+            "LEFT JOIN list_options AS lo ON lo.list_id = 'warehouse' AND " .
+            "lo.option_id = di.warehouse_id AND lo.activity = 1 " .
+            "WHERE " .
+            "s.drug_id = '$drug_id' AND " .
+            "lo.option_value IS NOT NULL AND lo.option_value = '$facility_id' AND " .
+            "s.sale_date > DATE_SUB(NOW(), INTERVAL $form_days DAY) " .
+            "AND s.pid != 0 $fwcond $uwcond";
```

Comment:
need to ensure $fwcond $uwcond stuff gets escaped

Reply:
Need to ensure escaping in below queries

Reply:
Right. $fwcond and $uwcond are manually escaped.